packages/apollo-voyager-conflicts/src/detectConflict.ts

```diff
@@ -1,36 +1,34 @@
-import { GraphQLError } from 'graphql'
+import debug from 'debug'
+import { CONFLICT_ERROR, CONFLICT_LOGGER } from './constants'
 import { ConflictResolutionData } from './handleConflict'
-const prefix = 'AgSync:'
-const CONFLICT_TYPE = prefix + 'DataConflict'
-const VALIDATION_TYPE = prefix + 'Validation'
+import { SyncServerError } from './SyncServerError'
 
-/**
- * Represents server side error
- */
-export class SyncServerError extends GraphQLError {
-
-  public type: string
-  public version: number
-  public data: any
-
-  constructor (message: string, data: any, type: string) {
-    super(message)
-    this.type = type || VALIDATION_TYPE
-    this.data = data
-    this.version = data.version
-  }
+interface ObjectState {
+  serverData: ConflictResolutionData,
+  clientData: ConflictResolutionData
+  detect (): SyncServerError | undefined
```

Comment:
`detect(server: ConflictResolutionData, client: ConflictResolutionData)`?
Missing `next` method to iterate to `next` state